cosmoz-default-tree.html

```diff
@@ -23,134 +32,121 @@
+			return this.getNodeByPathLocator(propertyValue);
 		}
 
-		return this._searchAllRoots(propertyName, propertyValue);
+		return this._searchNodeList(propertyName, propertyValue, exact, all, nodeList);
 	};
 
-	Cosmoz.DefaultTree.prototype.getPath = function (node) {
-		if (node === undefined) {
-			return;
-		}
-		return this._getPathByPathLocator(node.pathLocator);
+	/**
+	 * Returns the path String of the found Node(s).
+	 */
+	Cosmoz.DefaultTree.prototype.getPathByProperty = function (propertyName, propertyValue, exact, all, nodeList) {
+		var node = this.getNodeByProperty(propertyName, propertyValue, exact, all, nodeList);
+		return this.getPath(node.path, propertyName);
+	},
+
+	/**
+	 * Loops over a node list (or roots if not defined) and returns the concatenated results of search().
+	 */
+	Cosmoz.DefaultTree.prototype._searchNodeList = function (propertyName, propertyValue, exact, all, nodeList) {
+		var results = [];
+
+		// Defaults
+		exact = exact !== undefined ? exact : true;
+		all = all !== undefined ? all : false;
+		nodeList = nodeList ? nodeList : this._roots;
+
+		nodeList.forEach(function (node) {
+			results = results.concat(this.search(node, propertyName, propertyValue, exact, all));
+		}, this);
+			
+		return results;
 	};
 
-	Cosmoz.DefaultTree.prototype.getPathByProperty = function (propertyName, propertyValue) {
-		if (propertyName === undefined || propertyValue === undefined) {
-			return;
+	/**
+	 * Returns the Node of a given pathLocator.
+	 */
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator) {
+		if (!pathLocator) {
+			return this._treeData;
 		}
+		var	nodesOnPath = this.getNodesOnPath(pathLocator, this._treeData);
+		return nodesOnPath[nodesOnPath.length - 1];
+	};
 
-		if (propertyName === 'pathLocator') {
-			return this._getPathByPathLocator(propertyValue);
-		}
+	/**
+	 * Returns an Array of nodes on a given path.
+	 * If pathLocator is empty or not defined, the Root objects get returned.
+	 */
+	Cosmoz.DefaultTree.prototype.getNodesOnPath = function (pathLocator, nodes, separatorSign) {
+		if (!pathLocator) return this._roots;
+
+		// Defaults
+		separatorSign = separatorSign ? separatorSign : '.'
+
+		var path = pathLocator.split(separatorSign),
+			pathSegment = nodes ? nodes : this._treeData;
+		
+		return path.map(function (nodeKey) {
+			var node = pathSegment[nodeKey],
+				children = this.getChildren(node);
+			node['key'] = nodeKey;
+			if (node && children !== undefined && Object.keys(children).length > 0) {
+				pathSegment = children;
+			}
+			return node;
+		}, this);
+	},
 
-		var node = this._searchAllRoots(propertyName, propertyValue);
-		if (node !== undefined) {
-			return this._getPathByPathLocator(node.pathLocator);
-		}
+	/**
+	 * Returns a String (separated node comparison properties) representing the path.
+	 */
+	Cosmoz.DefaultTree.prototype.getPath = function (pathLocator, comparisonProperty, pathSeparator) {
```

Comment:
Since these are API methods, it should be clear that the old "getPathString" now is called "getPath" and the old "getPath" is now called "getNodesOnPath".
I actually think it would be good to keep the old method names during this refactor and rename the methods in a separate change.